platforms/documentation/docs/src/docs/userguide/dep-man/basics/declaring_repositories.adoc

```diff
@@ -0,0 +1,156 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[declaring-repositories]]
+= 3. Declaring repositories
+
+You can add repositories for your dependencies by configuring the repositories block in your `build.gradle(.kts)` file:
+
+[source,kotlin]
+.build.gradle.kts
+----
+repositories {
+    mavenCentral()  // <1>
+    maven {         // <2>
+        url = uri("https://company/com/maven2")
+    }
+    mavenLocal()    // <3>
+    flatDir {       // <4>
+        dirs("libs")
+    }
+}
+----
+<1> Public repository
+<2> Private/Custom repository
+<3> Local repository
+<4> File location
+
+Gradle can resolve dependencies from one or many repositories based on Maven, Ivy or flat directory formats.
+
+[[sec:declaring_public_repository]]
+== Declaring a public repository
+
+Organizations building software may want to leverage public binary repositories to download and consume open source dependencies.
+Popular public repositories include link:https://repo.maven.apache.org/maven2/[Maven Central] and the link:https://maven.google.com/[Google Android] repository.
+
+Gradle provides built-in shorthand notations for these widely-used repositories:
+
+[source,kotlin]
+.build.gradle.kts
+----
+repositories {
+    mavenCentral()
+    google()
+    gradlePluginPortal()
+}
+----
+
+Under the covers Gradle resolves dependencies from the respective URL of the public repository defined by the shorthand notation.
+All shorthand notations are available via the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.RepositoryHandler.html[RepositoryHandler] API.
+
+[[sec:declaring_custom_repository]]
+== Declaring a private or custom repository
+
+Most enterprise projects establish a binary repository accessible only within their intranet.
+In-house repositories allow teams to publish internal binaries, manage users and security, and ensure uptime and availability.
+
+Specifying a custom URL is useful for declaring less popular but publicly-available repositories.
+Repositories with custom URLs can be specified as Maven or Ivy repositories by calling the corresponding methods available on the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.RepositoryHandler.html[RepositoryHandler] API:
+
+[source,kotlin]
+.build.gradle.kts
+----
+repositories {
+    maven {
+        url = uri("https://maven-central.storage.apis.com")
+    }
+    ivy {
+        url = uri("https://github.com/ivy-rep/")
+    }
+}
+----
+
+[[sec:declaring_local_repository]]
+== Declaring a local repository
+
+Gradle can consume dependencies available in the link:https://maven.apache.org/guides/introduction/introduction-to-repositories.html[local Maven repository].
+Declaring this repository is beneficial for teams that publish to the local Maven repository with one project and consume the artifacts by Gradle in another project.
```

Comment:
This is generally not how we recommend doing this. This is very much "maven thinking" and using included builds is a much more preferred alternative to using mavenLocal.

It is however a common way for a Gradle build to depend on a maven build on the same machine